x/btcstaking/keeper/msg_server_test.go

```diff
@@ -403,6 +404,156 @@ func TestProperVersionInDelegation(t *testing.T) {
 	require.Equal(t, uint32(2), actualDel1.ParamsVersion)
 }
 
+// TestBtcStakingWithBtcReOrg creates an BTC staking delegation
+// with enough covenant signatures submitted to be considered ACTIVE.
+func TestBtcStakingWithBtcReOrg(t *testing.T) {
+	btcLightclientTipHeight := uint32(30)
+	// btc staking tx will be included at btcLightclientTipHeight - BTC confirmation depth
+	h, r, btcctParams, stakingTxHash := createActiveBtcDel(t, btcLightclientTipHeight)
+
+	// verifies the largest reorg without anything set
+	_, err := h.BTCStakingKeeper.LargestBtcReOrg(h.Ctx, &types.QueryLargestBtcReOrgRequest{})
+	require.EqualError(t, err, types.ErrLargestBtcReorgNotFound.Error())
+
+	// should not panic in end blocker since there is no reorg
+	require.NotPanics(t, func() {
+		_, err = btcstaking.EndBlocker(h.Ctx, *h.BTCStakingKeeper)
+		h.NoError(err)
+	})
+
+	// -------------- simulates a reorg of current tip - (BTC depth - 1) --------
+	// It should not panic in x/btcstaking end blocker as the reorg is at the limit allowed
+	// It should consider the BTC staking as PENDING, since the block depth was revoked
+	rBlockFrom, rBlockTo := datagen.GenRandomBTCHeaderInfo(r), datagen.GenRandomBTCHeaderInfo(r)
+	rBlockFrom.Height = btcLightclientTipHeight
+	rBlockTo.Height = btcLightclientTipHeight - (btcctParams.BtcConfirmationDepth - 1)
+	currLargestReorg := types.NewLargestBtcReOrg(rBlockFrom, rBlockTo)
+
+	err = h.BTCStakingKeeper.SetLargestBtcReorg(h.Ctx, currLargestReorg)
+	h.NoError(err)
+
+	// should not panic in end blocker since the reorg is less than the allowed
+	require.NotPanics(t, func() {
+		_, err = btcstaking.EndBlocker(h.Ctx, *h.BTCStakingKeeper)
+		h.NoError(err)
+	})
+
+	// checks the query with a reorg set
+	respLargestReOrg, err := h.BTCStakingKeeper.LargestBtcReOrg(h.Ctx, &types.QueryLargestBtcReOrgRequest{})
+	h.NoError(err)
+	require.Equal(t, respLargestReOrg.BlockDiff, currLargestReorg.BlockDiff)
+	require.Equal(t, respLargestReOrg.RollbackFrom.HashHex, rBlockFrom.ToResponse().HashHex)
+	require.Equal(t, respLargestReOrg.RollbackTo.HashHex, rBlockTo.ToResponse().HashHex)
+
+	// BTC staking tx is still seen as active rolling back to a block where the confirmation depth is less than btcctParams.BtcConfirmationDepth
+	// Should we handle this differently?
```

Comment:
@KonradStaniec 

Imagine the following:
1. btc staking tx included block 20
2. btc tip is 30
3. Msg create btc delegation is sent + cov sigs
4. BTC delegation is ACTIVE
5. rollback 9 blocks (30 -> 21) 

what should be the status of that BTC delegation? In my mind would make sense to be as pending again
But I am not sure in the complications this might have in other modules

For any reorg larger or equal than `btcctParams.BtcConfirmationDepth` case by case will need to be analyzed and verified how it should be revoked

Reply:
So in BTC rollback happens only if there is better chain presented and better chain means chain with more blocks. 

Therefore it this most probably after rollbacking `9` blocks, `10` or more different blocks will be inserted. This means that in case you presented the transaction included in block `20` will still be considered active. In other words, in recovery procedure we can only care about transactions in blocks which have been rollbacked.